lib/plugins/aws/deploy/compile/functions/index.js

```diff
@@ -114,6 +116,29 @@ class AwsCompileFunctions {
     _.merge(this.serverless.service.provider.compiledCloudFormationTemplate.Resources,
       newFunctionObject);
 
+    const newVersion = this.cfLambdaVersionTemplate();
+
+    const content = fs.readFileSync(artifactFilePath);
+    const hash = crypto.createHash('sha256');
+    hash.setEncoding('base64');
+    hash.write(content);
+    hash.end();
+
+    newVersion.Properties.CodeSha256 = hash.read();
+    newVersion.Properties.FunctionName = { Ref: functionLogicalId };
+
+    const normalSha = newVersion.Properties.CodeSha256.replace(/[^0-9a-z]/gi, '');
+
+    // use the SHA in the logical resource ID of the version because
+    // AWS::Lambda::Version resource will not support updates
+    const versionLogicalId = `${functionLogicalId}Version${normalSha}`;
```

Comment:
could move the logical ID generation into [the `naming.js` file](https://github.com/serverless/serverless/blob/master/lib/plugins/aws/lib/naming.js)?

Also I think this name is shorter and expressive enough: `{FunctionName}Version{Sha}` ... so we add the function name not the function logical ID. This should be enough/short identification because all function names are unique